third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp

```diff
@@ -219,12 +224,14 @@ class DotOpConversion : public ConvertTritonGPUOpToLLVMPattern<DotOp> {
   LogicalResult
   matchAndRewrite(DotOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    auto encodePrecision = [&](Type type) -> TritonGEN::PrecisionType {
+    auto encodePrecision =
+        [&](Type type, InputPrecisionAttr attr) -> TritonGEN::PrecisionType {
```

Comment:
```suggestion
        [](Type type, InputPrecisionAttr attr) -> TritonGEN::PrecisionType {
```
I think you can get rid of this since rewriter is no longer used in the lambda

Reply:
`bf16_ty`, `f16_ty` etc. are macros that expand to a call on `rewriter`, so we have to keep the capture as-is unfortunately.